include/robots/rc_car_common.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+namespace BoBRobotics {
+namespace Robots {
+namespace RCCar {
+
+#define RCCAR_SLAVE_ADDRESS 0x29
+
+enum class State : uint8_t
+{
+    RemoteControl,
+    I2CControl
+};
+
+enum class Command : uint8_t
+{
+    SetState,
+    ReadRemoteControl,
+    Drive
+};
+
+struct __attribute__((packed)) Movement
+{
+    int8_t speed, turn;
+};
+
+struct __attribute__((packed)) Message
+{
+    Command command;
+    union
+    {
+        State state;
+        Movement move;
+    };
```

Comment:
This packing seems a little sub-optimal as sizeof(State) == 1 but sizeof(Movement) == 2. Why not have:
```c++
enum class Command : uint8_t
{
    SetStateRemoteControl,
    SetStateI2CControl,
    ReadRemoteControl,
    Drive
};

struct __attribute__((packed)) Message
{
    Command command;
    Movement move;
}
```
save a byte **and** get rid of a union - what could be better :smile:

Reply:
Good idea!